stores/metadata.go

```diff
@@ -2496,3 +2557,103 @@ func (ss *SQLStore) processConsensusChangeContracts(cc modules.ConsensusChange)
 		height++
 	}
 }
+
+func buildMarkerExpr(db *gorm.DB, bucket, prefix, marker, sortBy, sortDir string) (markerExpr clause.Expr, orderBy clause.OrderBy, err error) {
+	// no marker
+	if marker == "" {
+		return exprTRUE, clause.OrderBy{}, nil
+	}
+
+	// for markers to work we need to order by object_id
+	orderBy = clause.OrderBy{
+		Columns: []clause.OrderByColumn{
+			{
+				Column: clause.Column{Name: "object_id"},
+				Desc:   false,
+			},
+		},
+	}
+
+	desc := strings.EqualFold(sortDir, api.ObjectSortDirDesc)
+	switch sortBy {
+	case "", api.ObjectSortByName:
+		if desc {
+			markerExpr = gorm.Expr("object_id < ?", marker)
+		} else {
+			markerExpr = gorm.Expr("object_id > ?", marker)
+		}
+	case api.ObjectSortByHealth:
+		// fetch marker health
+		var markerHealth float64
+		if marker != "" && sortBy == api.ObjectSortByHealth {
+			if err := db.
```

Comment:
Same thing. Is there a reason we can't use `objectHealth`?

Reply:
see https://github.com/SiaFoundation/renterd/pull/750#discussion_r1400732929